packages/grafana-ui/src/components/InteractiveTable/InteractiveTable.tsx

```diff
@@ -179,7 +179,8 @@ export function InteractiveTable<TableData extends object>({
 
   const tableHooks: Array<PluginHook<TableData>> = [useSortBy, useExpanded];
 
-  const paginationEnabled = pageSize > 0;
+  const multiplePages = data.length > pageSize;
+  const paginationEnabled = pageSize > 0 && multiplePages;
```

Comment:
This changes seemed sensible for my use-case but maybe it was intended to also show the pagination controls even if the data set is smaller than the page size?